mayastor/src/bdev/nexus/mod.rs

```diff
@@ -29,9 +35,79 @@ pub mod nexus_module;
 pub mod nexus_nbd;
 pub mod nexus_share;
 
+#[derive(Deserialize)]
+struct NexusShareArgs {
+    name: String,
+    protocol: String,
+    cntlid_min: u16,
+    cntlid_max: u16,
+}
+
+#[derive(Serialize)]
+struct NexusShareReply {
+    uri: String,
+}
+
 /// public function which simply calls register module
 pub fn register_module() {
-    nexus_module::register_module()
+    nexus_module::register_module();
+
+    jsonrpc_register(
+        "nexus_share",
+        |args: NexusShareArgs| -> Pin<Box<dyn Future<Output = Result<NexusShareReply>>>> {
+            // FIXME: shares bdev, not a nexus
+            let f = async move {
+                let proto = args.protocol;
+                if proto != "iscsi" && proto != "nvmf" {
+                    return Err(JsonRpcError {
+                        code: Code::InvalidParams,
+                        message: "invalid protocol".to_string(),
+                    });
+                }
+                if let Some(bdev) = Bdev::lookup_by_name(&args.name) {
+                    match proto.as_str() {
+                        "nvmf" => {
+                            bdev.share_nvmf(Some((args.cntlid_min, args.cntlid_max)))
```

Comment:
Generic question: are there any restrictions applied to (args.cntlid_min, args.cntlid_min) ? Should cntlid_min be always less than cntlid_min ? If user intentionally passes a lesser cntlid_min , will it break something ? Shall the values be different or so and shall we enforce some other sanity checks against the values passed to us ?

Reply:
Sanity checking is done in `spdk_nvmf_subsystem_set_cntlid_range`.